pkg/spdk/server.go

```diff
@@ -104,8 +105,41 @@ func (s *Server) monitoring() {
 			logrus.Info("SPDK Server: stopped monitoring replicas due to the context done")
 			done = true
 		case <-ticker.C:
-			if err := s.verify(); err != nil {
-				logrus.WithError(err).Errorf("SPDK Server: failed to verify and update replica cache, will retry later")
+			err := s.verify()
+			if err == nil {
+				break
+			}
+
+			logrus.WithError(err).Errorf("SPDK Server: failed to verify and update replica cache, will retry later")
+
+			if !jsonrpc.IsJSONRPCRespErrorBrokenPipe(err) {
+				break
+			}
+
+			running, err := util.IsSPDKTargetProcessRunning()
+			if err != nil {
+				logrus.WithError(err).Errorf("SPDK Server: failed to check spdk_tgt is running")
+				break
+			}
+			if running {
+				logrus.Info("SPDK Server: reconnecting to spdk_tgt")
+
+				cli, err := NewSPDKClient()
+				if err != nil {
+					logrus.WithError(err).Errorf("SPDK Server: failed to reconnect to spdk_tgt")
+					break
+				}
+
+				err = s.spdkClient.Close()
+				if err != nil {
+					logrus.WithError(err).Warnf("SPDK Server: failed to close spdk_tgt connection")
+				}
+				s.spdkClient = cli
+			} else {
```

Comment:
lock to protect `s.spdkClient`?

Reply:
Is a lock necessary?
Without a lock, some operations might use broken spdkClient, but they will use the new one next time.

Reply:
Yeah, since the client is already broken. It should be fine for now, but ideally, for a shared resource, it's going to change during the potential usage, it's better to protect it to prevent it has been changed in between. Hower, it's fine here.

Reply:
Refactored it and added a lock protection